builtin/credential/aws/client.go

```diff
@@ -0,0 +1,48 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/vault/logical"
+)
+
+// getClientConfig creates a aws-sdk-go config, which is used to create
+// client that can interact with AWS API. This reads out the secret key
+// and access key that was configured via 'config/client' endpoint and
+// uses them to create credentials required to make the AWS API calls.
+func getClientConfig(s logical.Storage) (*aws.Config, error) {
```

Comment:
This seems sucky.  If I'm running Vault in AWS, I probably want to use an IAM role to give Vault credentials to do this.  I'm more than willing to patch, but wondering if the fact that you're not even giving EC2 IAM roles a chance is intentional?